### ISSUE TITLE
Latest release wont install on windows 10

### ISSUE THREAD

**Issue by sirus20x6** [maharmstone/btrfs#296]
I get "data is invalid" when trying to install off the inf.
secure boot is off in the bios
driver signing is set to ignore in group policy

**Comment by ScottMorris** [comment]
I, too, get this error with 1.7.4 release.

![image](https://user-images.githubusercontent.com/404967/92986119-873df100-f486-11ea-904b-98d9f02530b1.png)

**Comment by hartmark** [comment]
+1

Get the same error

**Comment by hartmark** [comment]
It seems to be working now anyway after reboot after the latest updates

**Comment by maharmstone** [comment]
@sirus20x6, @ScottMorris - are there any hints in Event Viewer?

**Comment by maharmstone** [comment]
Also, what version of Windows are you on? (Try `winver`)

**Comment by itsTyrion** [comment]
Same thing here on a pretty fresh Win10 2004 install.

**Comment by FrederikLauber** [comment]
Hi, not getting an eror message on Windows 10 but my hard drives with btrfs are apparently unreadable. Is this related?

**Comment by ScottMorris** [comment]
I just tried it again and I can't reproduce the error, it just installed.  So weird. 

I looked back in the Event Logs and found this message,

```
Catalog Database (5724,D,50) Catalog Database: The database format feature version 9180 (0x23dc) could not be used due to the current database format 1568.20.0, controlled by the parameter 0x410022D8 (8920 | JET_efvAllowHigherPersistedFormat).
```

Here is the event XML,

```xml
- <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
- <System>
  <Provider Name="ESENT" /> 
  <EventID Qualifiers="0">642</EventID> 
  <Version>0</Version> 
  <Level>3</Level> 
  <Task>1</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2020-09-12T03:27:20.1992928Z" /> 
  <EventRecordID>7776</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="0" ThreadID="0" /> 
  <Channel>Application</Channel> 
  <Computer>Redacted-Desktop</Computer> 
  <Security /> 
  </System>
- <EventData>
  <Data>Catalog Database</Data> 
  <Data>5724,D,50</Data> 
  <Data>Catalog Database:</Data> 
  <Data>0x410022D8 (8920 | JET_efvAllowHigherPersistedFormat)</Data> 
  <Data>9180 (0x23dc)</Data> 
  <Data>1568.20.0</Data> 
  </EventData>
  </Event>
```

@maharmstone my Windows version is

![image](https://user-images.githubusercontent.com/404967/94325321-e6623200-ff6b-11ea-9470-40e6612c3a9d.png)

**Comment by inkubux** [comment]
I have a similar issue but the driver is installed and I see it in the device manager

However the disk is never mounted (It worked 2 days ago) but then a windows update broke it.

Not I see in the Events that the driver is not able to load the device and ends up with a status 0.

In the disk management util I see an Unallocated partiton with no Volumes

I did a check repair on the linux host and everything seems fine.

**Comment by itsTyrion** [comment]
A couple of days (and reboots) have passed, I retried installing.. and it worked - sort of.
I saw the small copy window on the top left (greetings from XP :D), heard the 'device plugged in' sound, followed by the Invalid data pop-up. To my surprise, I saw one of my Linux volumes (BTRFS) appear in the Explorer side bar.. it's actually mounted and accessible

**Comment by itsTyrion** [comment]
![Device Manager](https://user-images.githubusercontent.com/17730120/94599731-2ccbcf80-0291-11eb-9061-16ff6ac93f03.png)
![Content](https://user-images.githubusercontent.com/17730120/94599863-5c7ad780-0291-11eb-819b-76ba111b3abf.png)
I guess the structure is so weird because it's BTRFS (with timeshift)?

**Comment by ScottMorris** [comment]
@itsTyrion What you're seeing on `I:` are the top level subvolumes on your BTRFS drive.  `@` will be root (`/`), `@home` will contain your home directories (mounted at `/home/`), and `timeshift-btrfs` is used by timeshift to archive your TimeShift snapshots so you can move back and forth between them.

**Comment by itsTyrion** [comment]
> @itsTyrion What you're seeing on `I:` are the top level subvolumes on your BTRFS drive. `@` will be root (`/`), `@home` will contain your home directories (mounted at `/home/`), and `timeshift-btrfs` is used by timeshift to archive your TimeShift snapshots so you can move back and forth between them.

ty, figured so (I didn't set up this partition, it's a 2nd distro I'm just playing around with)

**Comment by itsTyrion** [comment]
![Any button on the Btrfs tab will lead to this error message, it's working tho](https://user-images.githubusercontent.com/17730120/94679050-c0e47800-031f-11eb-9660-1f94b44613de.png)

**Comment by Warrentheo** [comment]
I also experience this issue, I have Win10Pro 2004 running 19041.572...  I would be willing to send logs if requested...

**Comment by johnctharp** [comment]
![image](https://user-images.githubusercontent.com/29041576/99884984-6d8ff700-2bf7-11eb-9ee7-b7c244221224.png)
![image](https://user-images.githubusercontent.com/29041576/99885016-aaf48480-2bf7-11eb-97ef-b54d2160efe6.png)
2H02 here, same 'data is invalid' error, whether installing from the context menu with the .inf directly or using Chocolatey.

Want to say though, I do applaud the effort here; BTRFS isn't perhaps the most loved or exciting filesystem (see ZFS on Linux and Redhat tossing BRTFS to the curb...), but with respect to being an 'open' filesystem that is both modern and can be supported across a variety of operating systems, it's definitely a sight better than something based on FAT!

**Comment by maharmstone** [comment]
But it works when you restart, right?

> BTRFS isn't perhaps the most loved or exciting filesystem

Heresy!

**Comment by johnctharp** [comment]
> But it works when you restart, right?

Actually yes, thank you!

I did have to wrangle permissions a bit; Windows' default permissions (I'm testing the 'Downloads' Library) aren't very Unix-friendly, but a quick chown + chgrp fixed that with no apparent ill effects to the functionality of the Library.
> 
> > BTRFS isn't perhaps the most loved or exciting filesystem
> 
> Heresy!
I'm as surprised as anyone that Red Hat decided to go in a different direction; even more surprised to a degree that the direction they chose wasn't ZFS.

And removing support for BTRFS? Just... why. BTRFS is where neither ZFS nor anything else RH might choose as it's in the Linux kernel!

**Comment by jtcornett** [comment]
Just FYI, I've been using Paragon's Linux File Systems for Windows which supports read/write access ext2, ext3, and ext4.  I didn't realize it also has read access to btrfs and xfs.  I received that 'invalid data' error when installing.  After I reboot, still showing driver error in device manager.  When I discovered Paragon was reading btrfs, I guessed it was interfering with winbtrfs.  I was right.  It is working.  I like btrfs, but it has been 'different' in Linux shall we say (e.g. grub won't write to btrfs).  I'm sticking with it though.
